File: Anzac3/G/G.py
letters = input()

b_counts = [0]

cur = 0
for let in letters:
    if let == 'B':
        cur += 1
    else:
        cur -= 1
    b_counts.append(cur)

# Find max/min.

cmin = None
cmax = None

for i, amount in enumerate(b_counts):
    if cmin is None or amount < cmin[0]:
        cmin = (amount, i)
    if cmax is None or amount > cmax[0]:
        cmax = (amount, i)

s = min(cmax[1], cmin[1])
e = max(cmax[1], cmin[1])

print(s + 1, e)

File: Anzac3/G/G.cpp
#include <iostream>
#include <algorithm>
#include <cmath>
#include <string>

using namespace std;

int main()
{
    string s;
    cin >> s;
    int counts[100001];
    int cur = 0;
    counts[0] = cur;

    int minIndex = 0, maxIndex = 0;

    for (int i = 0; i < s.length(); i++)
    {
        if (s[i] == 'R') cur++;
        else if (s[i] == 'B') cur--;
        counts[i+1] = cur;
        if (counts[i+1] < counts[minIndex])
            minIndex = i + 1;
        if (counts[i+1] > counts[maxIndex])
            maxIndex = i + 1;
    }

    cout << min(minIndex, maxIndex) + 1 << ' ' << max(minIndex, maxIndex) << endl;
}
